packages/core/test/utils_test.js

```diff
@@ -1093,6 +1093,31 @@ describe("utils", () => {
         });
       });
 
+      it("should not create empty objects as the default value", () => {
+        const schema = {
+          type: "object",
+          required: [],
+          properties: {
+            value: {
+              type: "object",
+              required: ["a"],
+              properties: {
+                a: {
+                  type: "string",
+                },
+                b: {
+                  type: "string",
+                },
+                c: {
+                  type: "string",
+                },
+              },
+            },
+          },
+        };
+        expect(getDefaultFormState(schema, {})).eql({});
+      });
+
```

Comment:
What about a test for https://github.com/rjsf-team/react-jsonschema-form/issues/2708?

Reply:
Forgot to update the description. I review #2708 and the two issues are identical as far as I can tell.

Reply:
So -- that might be correct, but we should still add a test that explicitly tests for the scenario in #2708 -- i.e., tries to perform form validation and then ensures that the validation is working.